src/macros.jl

```diff
@@ -241,20 +256,20 @@ macro testitem(nm, exs...)
     end
     q = QuoteNode(exs[end])
     esc(quote
-        $store_test_item_setup(
-            $TestItem(
-                Ref(0), $nm, $tags, $default_imports, $setup, $retries,
-                $(String(__source__.file)), $(__source__.line),
-                $gettls(:__RE_TEST_PROJECT__, "."),
-                $q,
-                $TestSetup[],
-                Ref{Int}(0),
-                $DefaultTestSet[],
-                Ref{Int}(0),
-                $PerfStats[],
-                $ScheduledForEvaluation()
-            )
+        let ti = $TestItem(
+            Ref(0), $nm, $tags, $default_imports, $setup, $retries,
+            $(String(__source__.file)), $(__source__.line),
+            $gettls(:__RE_TEST_PROJECT__, "."),
+            $q,
         )
+            if $_quote || $gettls(:__RE_TEST_RUNNING__, false)::$Bool
+                $store_test_item_setup(ti)
+                ti
+            else # We are not in a `runtests` call, so we run the testitem eagerly.
+                $runtestitem(ti)
```

Comment:
Will this already default to the right eager logs mode and all the other normal REPL-experience settings?

Reply:
_Great_ question! The logs behaviour is correct but the `verbose_results` one isn't! will fix and add tests